src/checkers/inference/solver/frontend/TwoQualifiersLattice.java

```diff
@@ -0,0 +1,54 @@
+package checkers.inference.solver.frontend;
+
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+/**
+ * Special Lattice class for two qualifier type system.
+ * 
+ * @author jianchu
+ *
+ */
+public class TwoQualifiersLattice extends Lattice {
+
+    protected Set<AnnotationMirror> allTypes;
```

Comment:
We calls `qualHierarchy.getTypeQualifiers();` in `Lattice`, and that needs the type of `allTypes` to be `? extends AM`. That still causes problem in `public class TwoQualifiersLattice extends Lattice {
` even if we use `allTypes = Collections.unmodifiableSet(tempSet);`allTypes = Collections.unmodifiableSet(tempSet);